kobo/apps/superuser_stats/views.py

```diff
@@ -22,40 +31,227 @@ def _base_filename_to_full_filename(base_filename, username):
+        f'</a>'
+        f'</body></html>'
+    )
+
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def domain_report(request):
+    """
+    Generates a report which counts the amount of users
+    with the same email address domain
+    """
+    # Generate the file basename
+    today = timezone.now().date()
+    base_filename = 'domain-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        timezone.now().microsecond
+    )
+
+    # Get the date filters from the query and set defaults
+    start_date = request.GET.get(
+        'start_date',
+        f'{today.year}-{today.month}-{today.day}'
+    )
+    tomorrow = timezone.now() + timedelta(days=1)
+    end_date = request.GET.get(
+        'end_date',
+        f'{tomorrow.year}-{tomorrow.month}-{tomorrow.day}'
+    )
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_domain_report.delay(filename, start_date, end_date)
+
+    # Generate page text
+    url = f"{KOBOFORM_URL}{reverse('superuser_stats:domains_report')}"
+    template_ish = (
+        f'<html><head><title>Domains report</title></head>'
+        f'<body>Your report is being generated. Once finished, it will be '
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        f'If you receive a 404, please refresh your browser periodically until '
+        f'your request succeeds.<br><br>'
+        f'To select a date range, add a <code style="background: lightgray">?</code> at the end of the URL and set the '
+        f'<code style="background: lightgray">start_date</code> parameter to <code style="background: lightgray">YYYY-MM-DD</code> and/or the '
+        f'<code style="background: lightgray">end_date</code> parameter to <code style="background: lightgray">YYYY-MM-DD</code>.<br><br>'
+        f'<b>Example:</b><br>'
+        f'<a href="{url}?start_date=2020-01-31&end_date=2021-02-28">'
+        f'  {url}?start_date=2020-01-31&end_date=2021-02-28'
+        f'</a>'
+        f'<p>The default date range is for today, but submissions count will not be'
+        f' 0 unless it includes the range includes first of the month.</p>'
+        f'</body></html>'
+    )
+
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def forms_count_by_submission_report(request):
+    """
+    generates a report counting the number of forms within a
+    submission range
+    """
+    today = timezone.now().date()
+    base_filename = 'form-count-by-submissions-count-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        datetime.now().microsecond
+    )
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username
+    )
+    generate_forms_count_by_submission_range.delay(filename)
```

Comment:
Something seems off with this too:
- The end of the final line item's range is also `None`
- It doesn't make sense to have a range of `0-0`
- The total asset count doesn't seem to match what's on the server

Report:

Range | Count
-- | --
0-0 | 4
1-500 | 9
501-1000 | 0
1001-10000 | 0
10001-50000 | 0
50001-None | 0

Asset counts (unless these are the wrong things to filter on):
```
>>> len([a.has_deployment for a in Asset.objects.all()])
30
>>> Asset.objects.filter(_deployment_data__active=True).count()
19
```

Reply:
According to #3070

> A report that can be exportable in `.csv` format which includes a count of the number of projects over the last 12 months with the number of submissions in the following ranges

> 0
> 1-500
> 501-1 000
> 1001-10 000
> 10 000 - 50 000
> (50 001+ ?)

So the export matches what's on the server. But, it's not obvious that export only looks for last 12 months. 
@JacquelineMorrissette, IMO, we should add (again) some sort help/documentation in each HTML template to explain what exports are doing and what subset of data they export. (e.g. 12 months for this one). 

Having that said. If I read again the PR, I'm a little bit confused. 
Do we want to list all projects created in the last 12 months for each range (what's we are doing now) or do we want to list all projects from the beginning and aggregate the number of submissions for last 12 months. 
I think the latter would be more accurate. 

Let's say we have a project which has been created 2 years ago but still very active, we may want to see it appear in this report. cc @tinok?

Reply:
> Let's say we have a project which has been created 2 years ago but still very active, we may want to see it appear in this report.

Yes, I think we would want to see that project.

Reply:
This should not prevent merging based on our [internal discussion](https://chat.kobotoolbox.org/#narrow/stream/38-Kobo-usage-stats/topic/OCHA.20user.20stats/near/161089). Please keep #3070 open after the merge and update its description to indicate that we would want to include a project such as the one mentioned in Olivier's example.